proposals/dao/tip_121a.ts

```diff
@@ -0,0 +1,303 @@
+      sender: daiWhale,
+      fromInternalBalance: false,
+      recipient: daiWhale,
+      toInternalBalance: false
+    },
+    0,
+    '10000000000000000000000' // deadline
+  );
+
+  const postUserLusdBalance = await contracts.lusd.balanceOf(daiWhale);
+  const postUserDaiBalance = await contracts.dai.balanceOf(daiWhale);
+
+  const daiSpent = initialUserDaiBalance.sub(postUserDaiBalance);
+  expect(daiSpent).to.be.bignumber.equal(amountIn);
+
+  const lusdGained = postUserLusdBalance.sub(initialUserLusdBalance);
+  expect(lusdGained).to.be.bignumber.at.least(ethers.constants.WeiPerEther.mul(9_500));
+  expect(lusdGained).to.be.bignumber.at.most(ethers.constants.WeiPerEther.mul(11_000));
+
+  // Put in 10k DAI, got out ~10k LUSD
+  console.log('DAI spent: ', daiSpent / 1e18);
+  console.log('LUSD gained: ', lusdGained / 1e18);
+
+  // Execute OTC (impersonate Olympus multisig) to check it functions properly
+  const otcTribeAmount = '3753510000000000000000000';
+  const olympusSigner = await getImpersonatedSigner(addresses.olympusMultisig);
+  await contracts.tribe.connect(olympusSigner).approve(addresses.ohmEscrowUnwind, otcTribeAmount);
+  await contracts.ohmEscrowUnwind.connect(olympusSigner).swap();
+  const daoTribeBalanceAfterOtc = await contracts.tribe.balanceOf(addresses.core);
+  expect(daoTribeBalanceAfterOtc.sub(tribeInDaoTreasuryAfter)).to.be.equal(otcTribeAmount);
+  expect(await contracts.gOHMHoldingPCVDeposit.balance()).to.be.equal(0);
+
+  // Execute maxFeiWithdrawalGuard actions
+  let i = 0;
+  while (await contracts.maxFeiWithdrawalGuard.check()) {
+    const protecActions = await contracts.maxFeiWithdrawalGuard.getProtecActions();
+    const depositAddress = '0x' + protecActions.datas[0].slice(34, 74);
+    const depositLabel = getAddressLabel(addresses, depositAddress);
+    const withdrawAmountHex = '0x' + protecActions.datas[0].slice(138, 202);
+    const withdrawAmountNum = Number(withdrawAmountHex) / 1e18;
+    console.log('maxFeiWithdrawalGuard protec action #', ++i, depositLabel, 'withdraw', withdrawAmountNum);
+    await contracts.pcvSentinel.protec(contracts.maxFeiWithdrawalGuard.address);
+  }
+  // Execute fuseWithdrawalGuard actions
+  i = 0;
+  while (await contracts.fuseWithdrawalGuard.check()) {
+    const protecActions = await contracts.fuseWithdrawalGuard.getProtecActions();
+    const depositAddress = '0x' + protecActions.datas[0].slice(34, 74);
+    const depositLabel = getAddressLabel(addresses, depositAddress);
+    const withdrawAmountHex = '0x' + protecActions.datas[0].slice(138, 202);
+    const withdrawAmountNum = Number(withdrawAmountHex) / 1e18;
+    console.log('fuseWithdrawalGuard   protec action #', ++i, depositLabel, 'withdraw', withdrawAmountNum);
+    await contracts.pcvSentinel.protec(contracts.fuseWithdrawalGuard.address);
+  }
+};
+
+export { deploy, setup, teardown, validate };
+
+function getAddressLabel(addresses: NamedAddresses, address: string) {
+  for (const key in addresses) {
```

Comment:
`(Object.entries(addresses).find(entry => entry[0].toLowerCase() == address.toLowerCase()))[0]

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
Not important though for this pr.